cr-02-08-2022

Ami777 opened the pull request.

arakeee committed:
Merge pull request #4 from TheLocagus/feature/BE-2_database_config

Feature/be 2 database config

TheLocagus committed:
Merge pull request #8 from TheLocagus/main

Merge to not be develop behind main.

Commit:
Creating admin module, controller, service.
Added @nestjs/mapped-types to be able to using nest g resourse.
Added studentsToImportDTO.
Added importStudents method in adminService and add @Post to do it.

Commit:
Added generating uuid while adding new Student to db.
Changing id in StudentToImportDto as optional.
Added dto for HR, student datas and student's enums.

arakeee committed:
added entities with interfaces.

Commit:
Configurated sending emails.
Creating adding new recruiter.

Commit:
Fixed enums.

arakeee committed:
added entities with interfaces.

arakeee committed:
added entities with interfaces.

arakeee committed:
added entities with interfaces.

TheLocagus committed:
Removed yarn.lock to avoid conflict

Commit:
Merge branch 'develop' into feature/entities+interfaces

# Conflicts:
#	yarn.lock

Commit:
Merge branch 'develop' into feature/entities+interfaces

# Conflicts:
#	yarn.lock

Commit:
string[] to string in dto types

arakeee committed:
added password to student and recruiter and their interfaces

arakeee committed:
Merge remote-tracking branch 'origin/feature/entities+interfaces' into feature/entities+interfaces

TheLocagus committed:
Merge pull request #16 from TheLocagus/feature/entities+interfaces

Feature/entities+interfaces

Commit:
Resolving conflict

TheLocagus committed:
Merge pull request #17 from TheLocagus/feature/BE-admin_module

Feature/be admin module

Commit:
Added directory student-import

Commit:
Resolve some problems with names and relations after merge.

Commit:
Next fast hotfix.

Commit:
Added recruiter module, service, controller.
Added Post controller to send list of active users to frontend.
A little fixes like removing one option from enum.

TheLocagus committed:
Merge pull request #18 from TheLocagus/feature/BE-recruiter-module

Added recruiter module, service, controller.

Commit:
Changed endpoint, removed useless githubUser, added usefull id, changed title of available student.

Adam-DevPL committed:
installing needed packages for auth module

Adam-DevPL committed:
made some fixes for auth module

Adam-DevPL committed:
made auth module stuctures

Adam-DevPL committed:
add hash pwd util and decorator to get user from context

Adam-DevPL committed:
add dto and passport named strategies

Adam-DevPL committed:
add login functionality and cookie config

Adam-DevPL committed:
add logout functionality and cookie parser into main

Adam-DevPL committed:
enabling cors in main

Adam-DevPL committed:
modify mailerconfig to test sending emails and prepare test data - json file (insomnia)

Adam-DevPL committed:
modify gitignore to not sending yarn.lock

Adam-DevPL committed:
add storage function to point storage directory

Adam-DevPL committed:
modify endpoint to import students + validation of input data

Adam-DevPL committed:
little fix to recruiter

Adam-DevPL committed:
refactoring

Adam-DevPL committed:
fix, lack of registerToken in Recruiter

Adam-DevPL committed:
Merge pull request #19 from TheLocagus/feature/BE-8-auth-module

Feature/be 8 auth module

Adam-DevPL committed:
Merge pull request #20 from TheLocagus/fix/BE-9_11-import_students

Fix/be 9 1 import students

Adam-DevPL committed:
little fix to check if user account is active to be log in

Adam-DevPL committed:
fix: modify recruiter data when put in form new data (except email)

Adam-DevPL committed:
fix: modify student data when in the file there will be the same email

arakeee committed:
added student endpoints

Commit:
Added getOneRecruiterAndCompareToken and registerToken in recruiter.entity

Adam-DevPL committed:
Delete configMail.ts

it's secret!

Adam-DevPL committed:
Delete config.example.ts

No need to change example of config in develop

Adam-DevPL committed:
Merge pull request #26 from TheLocagus/feature/BE-get_one_recruiter

Added getOneRecruiterAndCompareToken and registerToken in recruiter.e…

TheLocagus committed:
Merge pull request #21 from TheLocagus/fix/BE-9_2-add_recruit

Fix/be 9 2 add recruit

TheLocagus committed:
Merge pull request #22 from TheLocagus/fix/BE-8-login_validation

Fix/be 8 login validation

TheLocagus committed:
Merge pull request #23 from TheLocagus/fix/BE-9.2-update_recruiter

Fix/be 9.2 update recruiter

TheLocagus committed:
Merge pull request #24 from TheLocagus/fix/BE-9.1-update_student

Fix/be 9.1 update student

Ami777 commented on data/testDataFile.json:
```diff
@@ -0,0 +1,90 @@
+[
```
Postarajcie się pisać nazwy plików `tak-jak-przyjeto-w-nest.json`

Ami777 commented on src/admin/admin.controller.ts:
```diff
@@ -0,0 +1,39 @@
+export class AdminController {
+  constructor(private readonly adminService: AdminService) {}
+
+  @Post('/import-students')
```
Ścieżki niezabezpieczone!

Adam-DevPL replied:
Nie zabezpieczone, bo jeszcze nie ma endpointów wszystkich.

Ami777 commented on src/admin/admin.service.ts:
```diff
@@ -0,0 +1,158 @@
+        throw new Error('No file or file format is different then JSON.');
+      }
+      const data = JSON.parse(
+        fs.readFileSync(
```
Absolutnie NIGDY nie używamy metod sync :)

Adam-DevPL replied:
poprawione, dzięki :)

Ami777 commented on src/admin/admin.service.ts:
```diff
@@ -0,0 +1,158 @@
+        }
+      }
+
+      fs.unlinkSync(
```
J/w

Adam-DevPL replied:
poprawione, dzięki :)

Ami777 commented on src/admin/admin.service.ts:
```diff
@@ -0,0 +1,158 @@
+        if (isStudentToImport(element)) {
+          students.push(element);
+        } else {
+          throw new Error('File has not proper data');
```
Czy przypadkiem jak tu będzie błąd to nie pominie usuwanie pliku?

Adam-DevPL replied:
Tutaj nie usunie, ale w dalej w cacth tak :)

Ami777 commented on src/auth/auth.service.ts:
```diff
@@ -0,0 +1,148 @@
+            return res.json({
+              error: 'Invalid login data or account not active!',
+            });
+          } else {
```
Ło Panie :D To trzeba uprościć - np. odwrócić tę logikę

Adam-DevPL replied:
To fakt :) Nie wygląda to ładnie. 
Najpierw skupiłem się na tym aby działało, a potem refactoring

Ami777 commented on src/auth/auth.service.ts:
```diff
@@ -0,0 +1,148 @@
+    const expiresIn = 60 * 60 * 24;
+    const accessToken = sign(
+      payload,
+      'jfhsjkfhkjsf878947289378978*&(*&*(&YUHJJBHGI&#Y78937893oUO*#UIOU#*U#*U*(#UOIJJhuHUH#iuhU*#&*(&#*(&#*&',
```
To do zmiennych (stałych)

Ami777 commented on src/auth/auth.service.ts:
```diff
@@ -0,0 +1,148 @@
+    let userWithThisToken = null;
+    do {
+      token = uuid();
+      if (user instanceof Admin) {
```
Bardzo ładne

Ami777 commented on src/auth/strategy/student.strategy.ts:
```diff
@@ -0,0 +1,36 @@
+    super({
+      jwtFromRequest: cookieExtractor,
+      secretOrKey:
+        'jfhsjkfhkjsf878947289378978*&(*&*(&YUHJJBHGI&#Y78937893oUO*#UIOU#*U#*U*(#UOIJJhuHUH#iuhU*#&*(&#*(&#*&',
```
J/w, powtarza się wielokrotnie. W dodatku kwestią bezpieczeństwa mieć to inne na serwerze docelowym

Ami777 commented on src/enums/student.enum.ts:
```diff
@@ -0,0 +1,14 @@
+export enum StudentTypeOfWorkEnum {
+  naMiejscu = 'Na miejscu',
```
Zdecydowanie nazwy nie powinny być po polsku - stringi jeszcze ewentualnie mogą.

Ami777 reviewed (commented):
Jest git :) W komentarzach kilka podpowiedzi

Commit:
Added new Post in admin controller and service just to add students without file. Just for tests.

Added Patch in recruiter to modify student's status.

Commit:
Added TODO about auth validation.

Commit:
Added getting all for-interview students for list on frontend.

Commit:
A little change in case 'employed'.

Commit:
Added StudentCvInterface.

Adam-DevPL committed:
fix: replace Sync method of readFile and add additional method

Adam-DevPL committed:
fix: add salt variable

Adam-DevPL committed:
fix: in login method, check type of user

Adam-DevPL committed:
fix: refactor helper function isStudentToImport

TheLocagus committed:
Merge pull request #32 from TheLocagus/fix/BE-8-login

Fix/be 8 login

TheLocagus committed:
Merge pull request #31 from TheLocagus/fix/BE-9.1-import_students-file-operations

Fix/be 9.1 import students file operations

TheLocagus committed:
Merge pull request #30 from TheLocagus/feature/BE-11.4-Patch-Recruiter

Feature/be 11.4 patch recruiter

TheLocagus committed:
Merge pull request #29 from TheLocagus/feature/BE-11.2-GetAllStudents_Recruiter

Feature/be 11.2 get all students recruiter

Adam-DevPL committed:
small fix

Adam-DevPL committed:
Merge pull request #33 from TheLocagus/destroy

small fix

Commit:
Fixed Removed useless endpoint and useless TODO.

Commit:
Added endOfReservation to StudentInterface and to ForInterviewStudentToListResponseInterface.

Added new column endOfReservation in Student table.

Added setting 10 days reservation after adding student to for-interview list.

Commit:
Added column avatar.

Commit:
Added two column ownCodeInTeamUrls and code ReviewUrls.

Commit:
Little fixes with student's statuses.

TheLocagus committed:
Merge pull request #34 from TheLocagus/fix/be-11.2/11.4

Fixed Removed useless endpoint and useless TODO.

TheLocagus committed:
Merge pull request #35 from TheLocagus/feature/be-new_column_with_date_reservation

Feature/be new column with date reservation

TheLocagus committed:
Merge branch 'develop' into fix/little-fixes

TheLocagus committed:
Merge pull request #36 from TheLocagus/fix/little-fixes

Little fixes with student's statuses.

arakeee committed:
Added student endpoints.

Commit:
Removed three column ownCodeInTeamUrls, codeReviewUrls and Avatar.

Commit:
Added @Get('cv/:id)' in recruiter.controller

Commit:
Removed avatar, ownCodeInTeam and codeReviewUrls in StudentCvInterface.

Commit:
Added endpoint getOneStudentCv.

TheLocagus committed:
Merge branch 'develop' into feature/BE_student_cv

TheLocagus committed:
Merge pull request #38 from TheLocagus/feature/BE_student_cv

Feature/be student cv

Commit:
Repairing types/student.ts after a failed conflict resolution attempt

TheLocagus committed:
Merge pull request #40 from TheLocagus/feature/BE_debug_student_cv

Repairing types/student.ts after a failed conflict resolution attempt

Commit:
Added three properties in StudentCvInterface

TheLocagus committed:
Merge pull request #41 from TheLocagus/feature/BE_debug_student_cv

Added three properties in StudentCvInterface

Commit:
Finished method for filters. Didn't resolve problem with pipes and filters yet.

arakeee committed:
Delete package.json

arakeee committed:
Delete package-lock.json

Commit:
Works with pipes!

TheLocagus committed:
Merge branch 'develop' into BE-filters

arakeee committed:
Added student endpoints.

Commit:
Transfered some enums to directory enums.
Validated filters on so many cases.

TheLocagus committed:
Merge pull request #43 from TheLocagus/second-fix

Second Fix for student endpoints

Commit:
A little change

Commit:
Resolving conflicts

TheLocagus committed:
Merge pull request #37 from TheLocagus/feature/student-endpoints

BE-10. Feature/student endpoints

Commit:
Correcting types.

Commit:
Repaired situation with removed package.json. Auch...

Commit:
Repaired types salary.

Commit:
Added @ to DTO, fixed paths and other stuff to work with FE

Commit:
Changed @Patch to @Get.

Adam-DevPL committed:
Merge pull request #44 from TheLocagus/BE-filters

Ready to test

Commit:
Added getAllWithSearchedPhrase endpoint in recruiter.

Commit:
Added new where to getAllStudents method in recruiter service.
Repaired imports of two enums: ContractType and TypeWork.

TheLocagus committed:
Merge branch 'develop' into feature/BE_searching

TheLocagus committed:
Merge branch 'develop' into feature/testing_user_registration

TheLocagus committed:
Merge pull request #45 from TheLocagus/feature/testing_user_registration

Added @ to DTO, fixed paths and other stuff to work with FE

TheLocagus committed:
Merge branch 'develop' into fix/BE_recruiter_getAll_and_some_type

TheLocagus committed:
Merge pull request #48 from TheLocagus/fix/BE_recruiter_getAll_and_some_type

Added new where to getAllStudents method in recruiter service.

Commit:
Added StatusResponse interface.
Added axios.
Changed datetime to date in endOfReservation entity type.
Added RecruiterActionOfStatusEnum.
Added avatarUrl to interface.

TheLocagus committed:
Merge pull request #47 from TheLocagus/feature/BE_searching

Added getAllWithSearchedPhrase endpoint in recruiter.

TheLocagus committed:
Merge pull request #49 from TheLocagus/feature/axios

Added StatusResponse interface.

arakeee committed:
Pagination for recruiter page.

arakeee committed:
Update recruiter.service.ts

arakeee committed:
Update recruiter.service.ts

arakeee committed:
Update recruiter.service.ts

Adam-DevPL committed:
add small changes - errors will be more accurate

Adam-DevPL committed:
add needed types

Adam-DevPL committed:
add some changes to Admin

Adam-DevPL committed:
Change structure files. Modify Student, add more validation, improve safety.

Adam-DevPL committed:
Make student more restful and minor changes.

arakeee committed:
Update recruiter.service.ts

Adam-DevPL committed:
Add pagination and small fix for Student PATCH

Adam-DevPL committed:
Next small fixes in Recruiter.

TheLocagus committed:
Merge pull request #51 from TheLocagus/BE-refactoring

Be refactoring

TheLocagus committed:
Merge branch 'develop' into feature/pagination

TheLocagus committed:
Merge pull request #50 from TheLocagus/feature/pagination

Feature/ BE.14 - pagination

Commit:
Hotfixed few import/type errors. Changed tsconfig.json

Commit:
Coordinating with pagination.

Adam-DevPL committed:
Forget password feature.

Adam-DevPL committed:
Merge branch 'develop' into BE-refactoring_v2

Commit:
Searching with pagination.

TheLocagus committed:
Merge pull request #52 from TheLocagus/BE-refactoring_v2

feature/forgot password

Commit:
Searching with pagination next changes.

TheLocagus committed:
Merge branch 'develop' into fix/coordinating_search_with_fe

TheLocagus committed:
Merge pull request #53 from TheLocagus/fix/coordinating_search_with_fe

Fix/coordinating search with fe

Commit:
Fixes.

Adam-DevPL committed:
On client demands, change that Student can be In Talk with several recruiters not only one.

Adam-DevPL committed:
The logic to recruiters can add students to talk, but now none of the students is blocked by recruiter to talk.

Commit:
Rename registeredStudentInfoEmailTemplate on registeredRecruiterInfoEmailTemplate in importRecruiters

TheLocagus committed:
Merge pull request #55 from TheLocagus/feature/BE_register_recruiter

Rename registeredStudentInfoEmailTemplate on registeredRecruiterInfoE…

Commit:
Coordinating with fe

TheLocagus committed:
Merge pull request #54 from TheLocagus/BE-14-client_changes

Be 14 client changes

TheLocagus committed:
Merge branch 'develop' into feature/pagging_filtring_searching

TheLocagus committed:
Merge pull request #56 from TheLocagus/feature/pagging_filtring_searching

Coordinating with fe

TheLocagus committed:
Merge pull request #57 from TheLocagus/develop

Develop

TheLocagus committed:
Merge pull request #58 from TheLocagus/main

Merge pull request #57 from TheLocagus/develop

arakeee committed:
Update README.md

arakeee committed:
Update README.md

Commit:
Changes to finish logic about recruiter registration.

TheLocagus committed:
Merge pull request #59 from TheLocagus/feature/set_recruiter_password

Changes to finish logic about recruiter registration.

Commit:
Get recruiter's data after login.

TheLocagus committed:
Merge pull request #60 from TheLocagus/feature/set_recruiter_password

Get recruiter's data after login.

Commit:
Fixed except statuses

TheLocagus committed:
Merge pull request #61 from TheLocagus/fix/trying_fix_paths

Fixed except statuses

Adam-DevPL committed:
Fix to new feature - many recruiters many students.

Adam-DevPL committed:
Add guards to Admin and Recruiter

Commit:
Synchronized student register data with BE.

TheLocagus committed:
Merge pull request #62 from TheLocagus/fix/BE-14-client_changes_new

Fix/be 14 client changes new

Adam-DevPL committed:
Add Recruiter and Student guards

Adam-DevPL committed:
Small fix

TheLocagus committed:
Merge pull request #63 from TheLocagus/feature/BE-Guards-add-protection-for-resource

Feature/be guards add protection for resource

TheLocagus committed:
Merge pull request #64 from TheLocagus/develop

Develop

Commit:
Small changes.
Added origin to enableCors.

Adam-DevPL committed:
Fix for logout

TheLocagus committed:
Merge pull request #65 from TheLocagus/fix/BE-Guards-add-protection-for-resource

Fix for logout

TheLocagus committed:
Merge pull request #66 from TheLocagus/develop

Develop

Commit:
Prepared config examples.

Commit:
Added prefix /api

Commit:
Added CORS origin to config.

TheLocagus committed:
Merge pull request #67 from TheLocagus/feature/api_routes_for_production

Feature/api routes for production

TheLocagus committed:
Merge branch 'main' into develop

TheLocagus committed:
Merge pull request #68 from TheLocagus/develop

Develop

Commit:
Fix

Commit:
Fix

Commit:
Added api/ to all controllers

TheLocagus committed:
Merge pull request #69 from TheLocagus/main

That is what happens when you forget change branch.

Commit:
Changes localhosts:3000 to correct address.

TheLocagus committed:
Merge pull request #70 from TheLocagus/main

Changes localhosts:3000 to correct address.

arakeee committed:
Update README.md

Commit:
Small changes in password-change/forget logic

TheLocagus committed:
Merge pull request #71 from TheLocagus/develop

Develop

TheLocagus committed:
Merge pull request #72 from TheLocagus/main

To not be develop behind main
